apps/meteor/server/services/messages/service.ts

```diff
@@ -121,11 +141,19 @@ export class MessageService extends ServiceClassInternal implements IMessageServ
 			Rooms.incMsgCountById(rid, 1),
 		]);
 
-		void notifyOnMessageChange({
-			id: result.insertedId,
-		});
+		if (!insertedId) {
+			throw new Error('Failed to save system message.');
+		}
+
+		const createdMessage = await Messages.findOneById(insertedId);
+		if (!createdMessage) {
+			throw new Error('Failed to find the created message.');
+		}
+
+		void notifyOnMessageChange({ id: createdMessage._id });
```

Comment:
since you just fetched the inserted message, we can optimize by passing the data to `notifyOnMessageChange` otherwise it will fetch it from DB again
```suggestion
		void notifyOnMessageChange({ id: createdMessage._id, data: createdMessage });
```